src/Microsoft.ML.Core/Data/ReadOnlyMemoryUtils.cs

```diff
@@ -0,0 +1,333 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.ML.Runtime.Internal.Utilities;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.ML.Runtime.Data
+{
+    public static class ReadOnlyMemoryUtils
+    {
+
+        /// <summary>
+        /// Compare equality with the given system string value.
+        /// </summary>
+        public static bool EqualsStr(string s, ReadOnlyMemory<char> memory)
+        {
+            Contracts.CheckValueOrNull(s);
+
+            if (s == null)
+                return memory.Length == 0;
+
+            if (s.Length != memory.Length)
+                return false;
+
+            return memory.Span.SequenceEqual(s.AsSpan());
+        }
+
+        public static IEnumerable<ReadOnlyMemory<char>> Split(ReadOnlyMemory<char> memory, char[] separators)
+        {
+            Contracts.CheckValueOrNull(separators);
+
+            if (memory.IsEmpty)
+                yield break;
+
+            if (separators == null || separators.Length == 0)
+            {
+                yield return memory;
+                yield break;
+            }
+
+            if (separators.Length == 1)
+            {
+                char chSep = separators[0];
+                for (int ichCur = 0; ;)
+                {
+                    int ichMinLocal = ichCur;
+                    for (; ; ichCur++)
+                    {
+                        Contracts.Assert(ichCur <= memory.Length);
+                        if (ichCur >= memory.Length)
+                        {
+                            yield return memory.Slice(ichMinLocal, ichCur - ichMinLocal);
+                            yield break;
+                        }
+                        if (memory.Span[ichCur] == chSep)
+                            break;
+                    }
+
+                    yield return memory.Slice(ichMinLocal, ichCur - ichMinLocal);
+
+                    // Skip the separator.
+                    ichCur++;
+                }
+            }
+            else
+            {
+                for (int ichCur = 0; ;)
```

Comment:
Can we use `MemoryExtensions.IndexOfAny` here to simply this code? And also it is getting vectorized in .NET Core 3.0.